extmod/modframebuf.c

```diff
@@ -115,7 +171,7 @@ static inline uint32_t getpixel(const mp_obj_framebuf_t *fb, int x, int y) {
 }
 
 STATIC void fill_rect(const mp_obj_framebuf_t *fb, int x, int y, int w, int h, uint32_t col) {
-    if (x + w <= 0 || y + h <= 0 || y >= fb->height || x >= fb->width) {
```

Comment:
> What was a purpose of the part that I have changed?

That's not a good question to ask ;-). (Akin to standing in a factory asking: what does that red button which I just pressed does?) Here it's clear that it checks whether a rectangle to be drawn and framebuffer's bounding rectangle intersect. If they don't, there's nothing to be drawn. If they do, there's something to draw, though perhaps a clipped part of the rectangle requested.

What's important is that this line has no relation to "adding GRAY4 format" whatsoever, so it shouldn't be a part of this change. And of course, before submitting a separate pull request to change it, you would first need to understand what it does (or otherwise just leave it alone). Take an example of https://github.com/micropython/micropython/pull/2751 - it's sad that @dpgeorge decided not to merge it, but it was done right - you addressed one issue in it, and it didn't contain unrelated changes. Please keep up doing it like that, not mix in different kind of changes like here.